src/game/phase/phase_pause.c

```diff
@@ -78,23 +84,27 @@ static int32_t Phase_Pause_DisplayRequester(
     const char *header, const char *option1, const char *option2,
     int16_t requested)
 {
-    static bool is_pause_text_ready = false;
-    if (!is_pause_text_ready) {
-        Requester_Init(&m_PauseRequester);
+    if (!m_IsTextReady) {
+        Requester_ClearTextstrings(&m_PauseRequester);
         Requester_SetSize(&m_PauseRequester, 2, -48);
         m_PauseRequester.requested = requested;
         Requester_SetHeading(&m_PauseRequester, header);
         Requester_AddItem(&m_PauseRequester, option1, 0);
         Requester_AddItem(&m_PauseRequester, option2, 0);
+        m_IsTextReady = true;
+        g_InputDB = (INPUT_STATE) { 0 };
+        g_Input = (INPUT_STATE) { 0 };
+    }
 
-        is_pause_text_ready = true;
+    // Don't allow back because it breaks the requestor items.
```

Comment:
I don't quite understand this scenario, can you elaborate?

Reply:
Oops forgot to answer this sorry for spams.

So `Requester_Display` clears the requester if `menu_back` is pressed.
```
    } else if (g_InputDB.menu_back) {
        Requester_ClearTextstrings(req);
        return -1;
    }
```

If you test the pause function in `develop`, you'll notice pressing `menu_back` will cause the requester to flash. That's because the text is cleared and has to be reinitialized. Since the player can't exit the pause requestor with `menu_back`, they have to select `Continue` or `No` on the second page, I decided to just zeroize the input if they press it. That fixes the requester flashing. Also, because we `malloc` now, I didn't want the requester malloc and freeing in this scenario.

Reply:
Thanks, this makes sense. Can we provide this context in the comment?